spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -408,152 +419,169 @@ def main(argv: Sequence[str]):
 
     # Actually do the computations
     im = Image(arguments.i)
-    data = im.data
-    for arg_name, arg_value in arguments.todo:
-        assert data is not None
-
-        if arg_name == "add":
-            if data.ndim == 4 and not arg_value:
-                # special case to sum a 4D volume across the t axis
-                data = np.sum(data, axis=3)
-            else:
+
+    # If volumewise is specified, treat 4D image as a set of 3D volumes
+    if arguments.volumewise:
+        data_list = [np.squeeze(arr, axis=3) for arr in np.array_split(im.data, im.data.shape[3], 3)]
+        dim = list(im.dim)
+        dim[3] = 1  # Replace 4D dim with 1 (indicating single 3D volume)
+        dim_list = [tuple(dim)] * len(data_list)
+    else:
+        data_list = [im.data]
+        dim_list = [im.dim]
+
+    data_out_list = []
+    for data, dim in zip(data_list.copy(), dim_list.copy()):
+        for arg_name, arg_value in arguments.todo:
+            assert data is not None
```

Comment:
Note that we're basically just indenting the entire if/else; GitHub is just showing the diff strangely.

(Encapsulating the if/else into a function might have made the diff a bit neater? But, it's nbd.)